tests/js-modules/modules.py

```diff
@@ -139,6 +139,25 @@ def test_bytecode_cache(network, args):
     return network
 
 
+@reqs.description("Test js engine options")
+def test_set_js_engine(network, args):
+    primary, _ = network.find_nodes()
+    # set js run time options
+    network.consortium.set_js_engine_options(
+        primary, max_heap_bytes=50 * 1024 * 1024, max_stack_bytes=1024 * 512
+    )
+    with primary.client("user0") as c:
+        r = c.get("/node/js_metrics")
+        body = r.body.json()
+        assert body["max_heap_size"] == 50 * 1024 * 1024
+        assert body["max_stack_size"] == 1024 * 512
+    # reset the heap and stack sizes to default values
+    network.consortium.set_js_engine_options(
+        primary, max_heap_bytes=100 * 1024 * 1024, max_stack_bytes=1024 * 1024
```

Comment:
If we don't have one already, it'd be good to have a test checking what happens when we exceed those limits, also to confirm that this is triggered around the amounts we expect.

Reply:
Sure, I created a TODO for this https://github.com/microsoft/CCF/issues/4371